test/gp/kernel.jl

```diff
@@ -192,6 +194,19 @@ using LinearAlgebra
                 differentiable_kernel_tests(k, ȳ, Ȳ, Ȳ_sq, X0, X1, X2; atol=1e-7, rtol=1e-7)
             end
         end
+        
+        @timedtestset "NeuralKernelNetwork" begin
+            k1 = 0.5 * stretch(EQ(), 0.1)
+            k2 = 1.0 * stretch(PerEQ(1.0), 0.2)
+            prim_layer = Primitive(k1, k2)
+
+            lin = LinearLayer(2, 2)
+            nn = Chain(lin, product)
+
+            nkn = NeuralKernelNetwork(prim_layer, nn)
+            differentiable_kernel_tests(nkn, ȳ, Ȳ, Ȳ_sq, X0, X1, X2; atol=1e-7, rtol=1e-7)
+        end
```

Comment:
This is a great start. Would be good to implement a few tests to make sure that the `NKN` recovers other kernels when it's meant to. eg. a `GP` with an `NKN` kernel with only a linear layer should be equivalent to a `GP` with a particular composite kernel. Having one or two tests that check that this holds (by, say, checking that the covariance matrices produced in both cases are almost equal) would be very helpful.